lib/libthr/thread/thr_stack.c

```diff
@@ -126,10 +126,7 @@ static char *last_stack = NULL;
 static inline size_t
 round_up(size_t size)
 {
-	if (size % _thr_page_size != 0)
-		size = ((size / _thr_page_size) + 1) *
-		    _thr_page_size;
-	return size;
+	return (size + _thr_page_size - 1) & ~(size_t)(_thr_page_size - 1);
```

Comment:
Any reason not to use `roundup2(size, _thr_page_size)`? It's either a compiler buildit or the same code.

Reply:
I forgot about `roundup2`.   Konstantin Belousov sent a diff to the mailing lists using that function.